presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PlanNodeDecorrelator.java

```diff
@@ -277,35 +276,40 @@ public PlanNodeDecorrelator(PlanNodeIdAllocator idAllocator, PlanVariableAllocat
 
         private Multimap<VariableReferenceExpression, VariableReferenceExpression> extractCorrelatedSymbolsMapping(List<Expression> correlatedConjuncts)
         {
-            // TODO: handle coercions and non-direct column references
             ImmutableMultimap.Builder<VariableReferenceExpression, VariableReferenceExpression> mapping = ImmutableMultimap.builder();
             for (Expression conjunct : correlatedConjuncts) {
+                // TODO: Handle non-direct column references for ArithmeticBinaryExpression, ArithmeticUnaryExpression, NotExpression, etc
```

Comment:
with the new approach for variable extraction do we now support these kinds of expressions?  would be good to add some tests.

Reply:
More UT tests added and yes current implementation covers decorrelation on all non-directive column references

Reply:
did you mean to update the PR? don't see any changes.

Reply:
yes PR is updated now, I was cleaning up some comments and UTs